### DISCUSSION TITLE
Can't connect dApp from mobile metamask

### DISCUSSION THREAD

**Discussion by Andryxatt** [WalletConnect/walletconnect-monorepo#1755] [Developer Support]
Hello evrybody, I em trying to make connection to metamask from PC and Mobile, i am using walletconnect, ethers js, metamask and 3 networks BSC, POLYGON and ETH. Everithing works fine on PC and Laptop, but when i em trying to connect my dapp from iphone, it opens app-store with metamask then i click open metamask but it doesn't connect to my dapp and walletconnect Sessions is empty! With the qrCode connection works correctly. What did I do wrong?
  // Configure chains for connectors to support
`  const { chains, provider } = configureChains(
    [mainnet, bsc, polygon],
    [
      infuraProvider({ infuraId }),
      publicProvider(),
    ]
  );`
` // Set up connectors
  const connectors = [
    new InjectedConnector({
      chains,
    }),
      new MetaMaskConnector({
       chains,
     }),
    new WalletConnectConnector({
      chains,
      options: {
        qrcode: false,
        rpc: {
          56: 'https://bsc-dataseed.binance.org/',
          137: 'https://rpc-mainnet.maticvigil.com/',
          1: `https://mainnet.infura.io/v3/${infuraId}`
        }
      }
    }),
  ]; const client = createClient({
    connectors,
    provider
  });
  const ethereumClient = new EthereumClient(client, chains)
<WagmiConfig client={client}>
        <App />
      </WagmiConfig>
      <Web3Modal projectId={projectId} ethereumClient={ethereumClient} />
`

**Comment by Vivienne-CPZ** [comment]
Have you solved this problem? I also have the same problem. When the mobile browser connects to the metaMask, it will open the app store. In the app store, it will open the metaMask. There is no signature and other information

**Reply by Andryxatt** [reply]
Not yet, can't connect iphone 7, 11, 13. But connect correctly on 12, 7SE, 12 Pro Max, 14 Pro Max, Android. When open metamask NO SIGNATURE to confirm connection (((

**Reply by Vivienne-CPZ** [reply]
Hongmeng OS also does not support

**Reply by Andryxatt** [reply]
I found only this information https://support.perawallet.app/en/article/resolving-walletconnect-issues-1tolptm/#3-issues-connecting-on-ios-with-mobile-safari

**Reply by Andryxatt** [reply]
Solution for this, to do few steps in metamask app on iPhone 
    Clear MetaMask privacy data: Settings > Security&Privacy > Clear privacy data
    Clear MetaMask browser cookies: Settings > Security & Privacy > Clear browser cookies
    Clear MetaMask browser history: Settings > Security & Privacy > Clear browser history

**Reply by Vivienne-CPZ** [reply]
After you clear the metaMask data, you can successfully connect? But we don't want users to set metaMask related configurations. Can we solve this problem in code?

**Comment by yanchain** [comment]
Not work,Is there any other effective method?
